org.adempiere.base/src/org/compiere/model/MInvoice.java

```diff
@@ -1943,7 +1943,42 @@ else if (!isSOTrx()
 		if (matchPO > 0)
 			info.append(" @M_MatchPO_ID@#").append(matchPO).append(" ");
 
+		// Generate and complete an allocation if is a credit memo linked to an invoice

```

Comment:
There is a problem here when the PaymentRule of the credit note is Cash, because a Payment and an Allocation are created in line 1768.

So, code here or there must take into account that fact.

Reply:
I don't know how to test (I don't use the Cash payment rule, but I think changing the first line of the patch 
`if (isCreditMemo() && getRelatedInvoice_ID() > 0) {`
by
`if (isCreditMemo() && getRelatedInvoice_ID() > 0 && !isPaid()) {`
should be enough.

wdyt ?

Reply:
> I don't know how to test (I don't use the Cash payment rule, but I think changing the first line of the patch

on pull requests you always **must** test in a vanilla idempiere



>  should be enough.

no idea if at that point the invoice arrives paid, you need to conduct tests for that - but I think as the condition is specific for cash payment rule maybe you need to check that, and the impact that causes

Reply:
When I said I don't use Cash payment rule, I meant I have no idea what is behind, what is the logic, how to use it... and therefore what needs to be tested.
I think this was discussed in last marketing commitee, but that could be useful to have a kind of 'community help needed' to help on analysis/test/....
(I agreee, is not the best place to discuss)